From 0d1065854848494f211c990ed26267565cc44647 Mon Sep 17 00:00:00 2001
From: Josh Blum <josh@joshknows.com>
Date: Thu, 6 Oct 2016 15:06:09 -0700
Subject: [PATCH 13/18] Support relocated install with VOLK_PREFIX env var

Some packaging systems such as snaps will install
the volk library to a dynamically chosen location.
The install script can set an evironment variable
so that the library reports the correct prefix.
---
 lib/constants.c.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/constants.c.in b/lib/constants.c.in
index 3839f53..a81c7cb 100644
--- a/lib/constants.c.in
+++ b/lib/constants.c.in
@@ -24,11 +24,14 @@
 #include <config.h>
 #endif
 
+#include <stdlib.h>
 #include <volk/constants.h>
 
 char*
 volk_prefix()
 {
+  const char *prefix = getenv("VOLK_PREFIX");
+  if (prefix != NULL) return (char *)prefix;
   return "@prefix@";
 }
 
-- 
2.11.0

